### PR TITLE
Bump Kani version to 0.46.0

### PR DESCRIPTION
**Pull request by feliperodri**
## What's Changed
* `modifies` Clauses for Function Contracts by @JustusAdam in https://github.com/model-checking/kani/pull/2800
* Fix ICEs due to mismatched arguments by @celinval in https://github.com/model-checking/kani/pull/2994. Resolves the following issues:
  * https://github.com/model-checking/kani/issues/2260
  * https://github.com/model-checking/kani/issues/2312
* Enable powf*, exp*, log* intrinsics by @tautschnig in https://github.com/model-checking/kani/pull/2996
* Upgrade Rust toolchain to nightly-2024-01-24 by @celinval @feliperodri @qinheping

**Full Changelog**: https://github.com/model-checking/kani/compare/kani-0.45.0...kani-0.46.0

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 and MIT licenses.

### CODE REVIEW

**Review by jaisnan — APPROVED** [review]
Thanks!

**Review by celinval — CHANGES_REQUESTED** [review]
Can you please include https://github.com/model-checking/kani/pull/2994 to the release notes. We might want to edit the message to include the issues that were fixed. Thanks

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,21 @@ This file contains notable changes (e.g. breaking changes, major changes, etc.)
...
 
+## [0.46.0]
+
+## What's Changed
+* `modifies` Clauses for Function Contracts by @JustusAdam in https://github.com/model-checking/kani/pull/2800
+* Fix ICEs due to mismatched arguments by @celinval in https://github.com/model-checking/kani/pull/2994. Resolves the following issues:
+  * https://github.com/model-checking/kani/issues/2260
+  * https://github.com/model-checking/kani/issues/2312
+  * https://github.com/model-checking/kani/issues/1365
+  * https://github.com/model-checking/kani/issues/1350
```
Comment (celinval) [inline CHANGELOG.md:15]:
```suggestion
  * https://github.com/model-checking/kani/issues/2260
  * https://github.com/model-checking/kani/issues/2312
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,19 @@ This file contains notable changes (e.g. breaking changes, major changes, etc.)
...
+## [0.46.0]
+
+## What's Changed
+* `modifies` Clauses for Function Contracts by @JustusAdam in https://github.com/model-checking/kani/pull/2800
+* Fix ICEs due to mismatched arguments by @celinval in https://github.com/model-checking/kani/pull/2994. Resolves the following issues:
+  * https://github.com/model-checking/kani/issues/2260
+  * https://github.com/model-checking/kani/issues/2312
+* Enable powf*, exp*, log* intrinsics by @tautschnig in https://github.com/model-checking/kani/pull/2996
+* Upgrade Rust toolchain to nightly-2024-01-23 by @celinval in https://github.com/model-checking/kani/pull/2983
+* Upgrade Rust toolchain to nightly-2024-01-24 by @feliperodri in https://github.com/model-checking/kani/pull/3008
```
Comment (celinval) [inline CHANGELOG.md:16]: We usually merge them too:
```suggestion
* Upgrade Rust toolchain to nightly-2024-01-24 by @celinval @feliperodri @qinheping
```

**Review by celinval — APPROVED** [review]

**Pull request merged**